code/game/objects/items/chainsaw.dm

```diff
@@ -0,0 +1,87 @@
+
+// CHAINSAW
+/obj/item/chainsaw
+	name = "chainsaw"
+	desc = "A versatile power tool. Useful for limbing trees and delimbing humans."
+	icon_state = "chainsaw_off"
+	lefthand_file = 'icons/mob/inhands/weapons/chainsaw_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/chainsaw_righthand.dmi'
+	flags_1 = CONDUCT_1
+	force = 13
+	var/force_on = 24
+	w_class = WEIGHT_CLASS_HUGE
+	throwforce = 13
+	throw_speed = 2
+	throw_range = 4
+	custom_materials = list(/datum/material/iron=13000)
+	attack_verb = list("sawed", "torn", "cut", "chopped", "diced")
+	hitsound = "swing_hit"
+	sharpness = IS_SHARP
+	actions_types = list(/datum/action/item_action/startchainsaw)
+	tool_behaviour = TOOL_SAW
+	toolspeed = 0.5
+	var/on = FALSE
+	var/wielded = FALSE // track wielded status on item
```

Comment:
anywhere where this is needed, you're probably making a mistake. don't make a var/wielded

Reply:
Based on Ninja's comments and the fact the wielded status is used in logic this is staying where it is.

Reply:
I told you as well to use the update_icon signal

Reply:
> Use the wield and unwielded signals to keep track of the state. ~ninjanomnom

Reply:
wielded status is used in /obj/item/chainsaw/get_dismemberment_chance()

Reply:
For some things yeah, I don't know why this is the comment I responded on in regards to this though